src/pod/babashka/aws/impl/aws/credentials.clj

```diff
@@ -0,0 +1,168 @@
+                                                   "default"))))
+  ([jvm-props profile-name]
+   (with-system-properties jvm-props
+     (-profile-credentials-provider+ jvm-props profile-name (or (io/file (u/getenv "AWS_CREDENTIAL_PROFILES_FILE"))
+                                                                (io/file (u/getProperty "user.home") ".aws" "credentials")))))
+  ([_jvm-props profile-name ^java.io.File f]
+   (create-provider
+    (creds/auto-refreshing-credentials
+     (reify creds/CredentialsProvider
+       (fetch [_]
+         (when (.exists f)
+           (try
+             (let [profile (get (config/parse f) profile-name)
+                   profile (if-let [cmd (get profile "credential_process")]
+                             (merge profile (get-credentials-via-cmd cmd))
+                             profile)]
+               (creds/valid-credentials
+                {:aws/access-key-id     (get profile "aws_access_key_id")
+                 :aws/secret-access-key (get profile "aws_secret_access_key")
+                 :aws/session-token     (get profile "aws_session_token")}
+                "aws profiles file"))
+             (catch Throwable t
+               (log/error t "Error fetching credentials from aws profiles file")
+               {})))))))))
+
+(def http-client pod.babashka.aws.impl.aws/http-client)
+
+(defn -default-credentials-provider [jvm-props]
+   (with-system-properties jvm-props
+     (create-provider (creds/default-credentials-provider @http-client))))
+
+(extend-protocol creds/CredentialsProvider
+  clojure.lang.PersistentArrayMap
+  (fetch [m]
+    (creds/fetch (get-provider m))))
+
+;;; Pod Client
+
+(defn -fetch [provider]
+  (when-let [provider (get-provider provider)]
+    (creds/fetch provider)))
+
+(def lookup-map
+  {'-fetch -fetch
+   '-basic-credentials-provider -basic-credentials-provider
+   '-system-property-credentials-provider -system-property-credentials-provider
+   '-profile-credentials-provider -profile-credentials-provider
+   '-profile-credentials-provider+ -profile-credentials-provider+
+   '-default-credentials-provider -default-credentials-provider})
+
+(def describe-map
+  `{:name pod.babashka.aws.credentials
+    :vars
+    ~(conj (mapv (fn [[k _]]
+                   {:name k})
+                 lookup-map)
+           {:name "fetch"
+            :code (pr-str
+                   '(defprotocol CredentialsProvider
+                      (fetch [_])))}
+           {:name "map->Provider"
+            :code (pr-str
+                   '(defrecord Provider [provider-id]
+                      CredentialsProvider
+                      (fetch [provider]
+                        (-fetch provider))))}
+           {:name "profile-credentials-provider"
+            :code (pr-str
+                   '(defn profile-credentials-provider [& args]
+                      (map->Provider (apply -profile-credentials-provider (cons (System/getProperties) args)))))}
```

Comment:
Can you restrict sending only the properties that are relevant to the pod?

```
user=> (count (keys (System/getProperties)))
56
```

This saves a lot of data over the wire.

Reply:
Good point, implemented here 25b0dd3